modules/index/src/main/java/org/apache/ignite/internal/index/IndexChooser.java

```diff
@@ -158,7 +160,7 @@ public void close() {
      */
     List<CatalogIndexDescriptor> chooseForRwTxUpdateOperation(int catalogVersion, int tableId) {
         return inBusyLock(busyLock, () -> {
-            List<CatalogIndexDescriptor> tableIndexes = catalogService.indexes(catalogVersion, tableId);
+            List<CatalogIndexDescriptor> tableIndexes = collectAvailableAndBuildingIndexes(catalogService.indexes(catalogVersion, tableId));
```

Comment:
We must write to indices that were REGISTERED at the tx start as well

Reply:
fix it